WooCommerce/Classes/ViewRelated/Orders/Order Details/Receipts/ReceiptEmail/ReceiptEmailViewModel.swift

```diff
@@ -34,20 +46,57 @@ final class ReceiptEmailViewModel: ObservableObject {
                 self.state = .idle
                 switch result {
                 case let .success(order):
+                    self.analytics.track(event: .InPersonPayments.receiptEmailSuccess(
+                        countryCode: self.countryCode,
+                        cardReaderModel: self.cardReaderModel,
+                        source: .api)
+                    )
+                    self.order = order
                     self.state = .success
                     DispatchQueue.main.asyncAfter(deadline: .now() + 0.6) {
-                        self.onDismiss(order)
+                        self.dismiss = true
                     }
                 case let .failure(error):
                     DDLogError("Sending email receipt failed: \(error.localizedDescription)")
+                    self.analytics.track(event: .InPersonPayments.receiptEmailFailed(
+                        error: error,
+                        countryCode: self.countryCode,
+                        cardReaderModel: self.cardReaderModel,
+                        source: .api)
+                    )
                     self.noticePresenter.enqueue(notice: Notice(title: Localization.errorNotice, feedbackType: .error))
-                 }
+                }
             }
         }
 
         self.state = .loading
         stores.dispatch(action)
     }
+
+    func onAppear() {
+        analytics.track(event: .InPersonPayments.receiptEmailTapped(
+            countryCode: countryCode,
+            cardReaderModel: cardReaderModel,
+            source: .api)
+        )
+    }
+
+    func onDisappear() {
+        if state != .success {
+            analytics.track(event: .InPersonPayments.receiptEmailCanceled(
+                countryCode: countryCode,
+                cardReaderModel: cardReaderModel,
+                source: .api)
+            )
+        }
+
+        self.noticePresenter.presentingViewController = nil
```

Comment:
super nit: self probably not needed (same in `onCancel`)
```suggestion
        noticePresenter.presentingViewController = nil
```